core/src/main/java/nl/nn/adapterframework/doc/model/FrankElement.java

```diff
@@ -1,27 +1,225 @@
+		configChildLookup = new HashMap<>();
+		for(ConfigChild c: children) {
+			ConfigChildKey key = new ConfigChildKey(c);
+			if(configChildLookup.containsKey(key)) {
+				log.warn(String.format("Different config children of Frank element [%s] have the same key", fullName));
+			} else {
+				configChildLookup.put(key, c);
+			}
+		}
+	}
+
+	public List<ConfigChild> getConfigChildren(Predicate<? super ConfigChild> filter) {
+		return configChildren.stream().filter(filter).collect(Collectors.toList());
+	}
+
+	ElementChild findElementChildMatch(ElementChild elementChild) {
+		if(elementChild instanceof FrankAttribute) {
+			return findAttributeMatch((FrankAttribute) elementChild);
+		} else if(elementChild instanceof ConfigChild) {
+			return findConfigChildMatch((ConfigChild) elementChild);
+		} else {
+			throw new IllegalArgumentException(String.format(
+					"Expected a FrankAttribute or ConfigChild, but got a [%s]",
+					elementChild.getClass().getName()));
+		}
+	}
+
+	FrankAttribute findAttributeMatch(FrankAttribute attribute) {
+		return attributeLookup.get(attribute.getName());
+	}
+
+	ConfigChild findConfigChildMatch(ConfigChild configChild) {
+		return configChildLookup.get(new ConfigChildKey(configChild));
+	}
+
+	public <T extends ElementChild> FrankElement getNextAncestor(Predicate<ElementChild> childFilter, Class<T> kind) {
+		FrankElement ancestor = parent;
+		while((ancestor != null) && (ancestor.getChildren(childFilter, kind).size() == 0)) {
+			ancestor = ancestor.getParent();
+		}
+		return ancestor;
+	}
+
+	public <T extends ElementChild> List<ElementChild> getChildren(Predicate<ElementChild> selector, Class<T> kind) {
+		List<ElementChild> result = new ArrayList<>();
+		if(kind.isAssignableFrom(FrankAttribute.class)) {
+			result.addAll(getAttributes(selector));
+		}
+		if(kind.isAssignableFrom(ConfigChild.class)) {
+			result.addAll(getConfigChildren(selector));
+		}
+		return result;
+	}
+
+	public void walkCumulativeAttributes(
+			CumulativeChildHandler<FrankAttribute> handler,
+			Predicate<ElementChild> childSelector,
+			Predicate<ElementChild> childRejector) {
+		new AncestorChildNavigation<String, FrankAttribute>(
+				handler, childSelector, childRejector, FrankAttribute.class) {
+			@Override
+			String keyOf(ElementChild child) {
+				return ((FrankAttribute) child).getName();
+			}
+		}.run(this);
+	}
+
+	public void walkCumulativeConfigChildren(
+			CumulativeChildHandler<ConfigChild> handler,
+			Predicate<ElementChild> childSelector,
+			Predicate<ElementChild> childRejector) {
+		new AncestorChildNavigation<ConfigChildKey, ConfigChild>(
+				handler, childSelector, childRejector, ConfigChild.class) {
+			@Override
+			ConfigChildKey keyOf(ElementChild child) {
+				return new ConfigChildKey((ConfigChild) child);
+			}
+		}.run(this);		
+	}
+
+	public void addAliasSource(ConfigChild aliasSource) {
+		aliasSources.add(aliasSource);
+	}
+
+	public String getAlias() {
```

Comment:
Wat is een alias in dit geval? Zou je dat hier kunnen documenteren?
-> ik zag later dat je dit al gedocumenteerd heb bij DocWriterNew. Toch graag hier ook twee regels, en evt. verwijzen naar DocWriterNew.

Reply:
Gedaan.